### PR TITLE
New personal statement journey

### PR DESCRIPTION
**Pull request by inulty-dfe**
## Context

Implement the new Personal Statement Journey.

With continuous applications, there is a new personal statement per application choice. The personal statement will be copied from the `ApplicationForm#becoming_a_teacher` when the application choice is submitted. Candidates must update the `ApplicationForm#becoming_a_teacher` field before each application choice submission in order to customise the personal statement for that application choice.

Before continuous applications, when an application form was `submitted` the personal statement could not be reviewed or edited. Now in continuous applications, the application form becomes submitted after the first application choice is submitted. But the candidate needs access to edit the personal statement in order to customise it for the next application choice.

We allow the candidate to edit personal statement if the application form is submitted and the application form is continuous applications.

### Vendor API

All versions of the Vendor API will return the `personal_statement` that exists on the `ApplicationChoice` model rather than the values of `becoming_a_teacher` and `subject_knowledge` that exist on the `ApplicationForm` model for applications in 2024 recruitement cycle (continuous applications)

## Link to Trello card

https://trello.com/c/9XnY0FJ3/38-ca-build-the-new-personal-statement-journey

Partially: https://trello.com/c/tUQa5Eus/36-ca-allow-application-form-sections-to-be-editable

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by tomas-stefano — APPROVED** [review]

**Comment by JR-G** [comment]
We'll also need to update `app/components/shared/personal_statement_component.html.erb`, which is used on Manage, to render the application choice personal statement if continuous applications is enabled

**Comment by inulty-dfe** [comment]
## Screenshots

The Provider interface has changed. 
In the support interface a provider will only see one application choice at a time. We want to display the personal statement belonging to the application choice relevant to the provider.

|Personal Statement on Form|Personal Statement on Choice|
|---|---|
|form|choice|

File: app/views/provider_interface/application_choices/show.html.erb [hunk 1]
```diff
@@ -68,7 +68,11 @@
 
 <%= render LanguageSkillsComponent.new(application_form: @application_choice.application_form) %>
 
-<%= render PersonalStatementComponent.new(application_form: @application_choice.application_form) %>
+<% if @application_choice.application_form.continuous_applications? %>
+  <%= render ChoicePersonalStatementComponent.new(application_choice: @application_choice) %>
+<% else %>
+  <%= render PersonalStatementComponent.new(application_form: @application_choice.application_form) %>
+<% end %>
```
Comment (tomas-stefano) [inline app/views/provider_interface/application_choices/show.html.erb:75]: 👍

Comment (tomas-stefano) [reply app/views/provider_interface/application_choices/show.html.erb:75]: Is there a reason behind creating a new component? I am okay by the way of using the new component I just want to understand the reasoning. The good thing about creating new ones is that we can delete the old ones :). Is that the reason?

File: app/components/shared/choice_personal_statement_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<section id="personal-statement-section" class="app-section govuk-!-width-two-thirds">
+  <h2 class="govuk-heading-m govuk-!-font-size-27" id="personal-statement"><%= t('personal_statement.title') %></h2>
+
+    <% if @editable %>
```
Comment (tomas-stefano) [inline app/components/shared/choice_personal_statement_component.html.erb:4]: Is this component being used only on provider interface for now, correct? If so we might do not need this if and the block inside. If is used on candidate interface or support, ignore this comment.

File: app/views/provider_interface/application_choices/show.html.erb (hunk 1 above)
Comment (JR-G) [reply app/views/provider_interface/application_choices/show.html.erb:75]: Iain and I spoke about this yesterday – the `PersonalStatementComponent` is being used on both the support and provider UI. Changing this to render just the application choice personal statement makes sense for Manage, but not such a great experience for support.

We'll turn our attention to the support UI once we've launched in the new cycle, we'll update the support application choice component to include the personal statement content

File: app/components/shared/choice_personal_statement_component.html.erb (hunk 2 above)
Comment (inulty-dfe) [reply app/components/shared/choice_personal_statement_component.html.erb]: Updated the component by removing `editable` from the signature and moving the component to the `ProviderInterface` namespace. Also added a simple spec

**Review by tomas-stefano — APPROVED** [review]

**Review by JR-G — APPROVED** [review]

**Pull request merged**